src/Text/Layout/Table/Cell.hs

```diff
@@ -10,68 +13,169 @@ import Text.Layout.Table.Spec.OccSpec
             []      -> Nothing
             _ : rs' -> Just $ length rs'
 
+    buildCellView = buildCellViewLRHelper stringB drop (\n s -> zipWith const s $ drop n s)
     buildCell = stringB
 
 instance Cell T.Text where
-    dropLeft = T.drop
-    dropRight = T.dropEnd
     visibleLength = T.length
     measureAlignment p xs = case T.break p xs of
         (ls, rs) -> AlignInfo (T.length ls) $ if T.null rs
             then Nothing
             else Just $ T.length rs - 1
 
+    buildCellView = buildCellViewLRHelper textB T.drop T.dropEnd
     buildCell = textB
 
+-- | Construct 'buildCellView' from a builder function, a function for
+-- trimming from the left, and a function for trimming from the right.
+--
+-- Used to define instances of 'Cell'.
+buildCellViewLRHelper :: StringBuilder b
+                      => (a -> b)  -- ^ Builder function for 'a'.
+                      -> (Int -> a -> a)  -- ^ Function for trimming on the left.
+                      -> (Int -> a -> a)  -- ^ Function for trimming on the right.
+                      -> CellView a
+                      -> b
+buildCellViewLRHelper build trimL trimR =
+    buildCellViewHelper build build build trimL trimR (\l r -> trimL l . trimR r)
+
+-- | Construct 'buildCellView' from a builder function, and a function for
+-- trimming from the left and right simultaneously.
+--
+-- Used to define instanced of 'Cell'.
+buildCellViewBothHelper :: StringBuilder b
+                        => (a -> b)  -- ^ Builder function for 'a'.
+                        -> (Int -> Int -> a -> a)  -- ^ Function for trimming on the left and right simultaneously.
+                        -> CellView a
+                        -> b
+buildCellViewBothHelper build trimBoth =
+    buildCellViewHelper build build build (`trimBoth` 0) (trimBoth 0) trimBoth
+
+-- | Construct 'buildCellView' from builder functions, and trimming functions.
+--
+-- Used to define instances of 'Cell'.
+buildCellViewHelper :: StringBuilder b
+                    => (a -> b)  -- ^ Builder function for 'a'.
+                    -> (a' -> b)  -- ^ Builder function for the result of trimming 'a'.
+                    -> (a'' -> b)  -- ^ Builder function for the result of trimming 'a' twice.
+                    -> (Int -> a -> a')  -- ^ Function for trimming on the left.
+                    -> (Int -> a -> a')  -- ^ Function for trimming on the right.
+                    -> (Int -> Int -> a -> a'')  -- ^ Function for trimming on the left and right simultaneously.
+                    -> CellView a
+                    -> b
+buildCellViewHelper build build' build'' trimL trimR trimBoth (CellView a l r) =
+    case (compare l 0, compare r 0) of
+        (GT, GT) -> spacesB l <> build a <> spacesB r
+        (GT, LT) -> spacesB l <> build' (trimR (-r) a)
+        (GT, EQ) -> spacesB l <> build a
+        (LT, GT) -> build' (trimL (-l) a) <> spacesB r
+        (LT, LT) -> build'' $ trimBoth (-l) (-r) a
```

Comment:
I have to admit that I am not a huge fan of unary minus.  I think it is hard to read, especially if there is no space, and it looks alien to me.  I tend to use `negate` in its place.

Thus, I would ask you to insert space for consistency with code below or use `negate`.